<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SegmenterAdvancedOptionsWizardPageForm</class>
 <widget class="QWizardPage" name="SegmenterAdvancedOptionsWizardPageForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>412</width>
    <height>325</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Segmenter Advanced Options</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <property name="leftMargin">
    <number>9</number>
   </property>
   <property name="topMargin">
    <number>9</number>
   </property>
   <property name="rightMargin">
    <number>9</number>
   </property>
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Advanced Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Block Options</string>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_10">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_9">
             <item row="0" column="0">
              <widget class="QCheckBox" name="m_enableBlockProcessingcheckBox">
               <property name="toolTip">
                <string>The original raster will be splitted into small blocks, each one will be segmented independently and the result will be merged at the end</string>
               </property>
               <property name="text">
                <string>Enable block processing</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <layout class="QGridLayout" name="gridLayout_7">
               <item row="0" column="0">
                <layout class="QGridLayout" name="gridLayout_6">
                 <item row="0" column="0">
                  <layout class="QHBoxLayout" name="horizontalLayout_4">
                   <item>
                    <widget class="QLabel" name="inputDataSetLabel_4">
                     <property name="minimumSize">
                      <size>
                       <width>170</width>
                       <height>0</height>
                      </size>
                     </property>
                     <property name="text">
                      <string>Maximum block size:</string>
                     </property>
                     <property name="alignment">
                      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="m_maximumBlockSizeLineEdit">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="toolTip">
                      <string>The input image will be split into blocks with this width for processing, This parameter tells the maximum block lateral size (width or height), the default: 0 - the size will be defined following the current system resources and physical processors number)</string>
                     </property>
                     <property name="text">
                      <string>0</string>
                     </property>
                     <property name="alignment">
                      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="1" column="0">
                  <layout class="QGridLayout" name="gridLayout">
                   <item row="0" column="0">
                    <widget class="QLabel" name="label_3">
                     <property name="minimumSize">
                      <size>
                       <width>170</width>
                       <height>0</height>
                      </size>
                     </property>
                     <property name="text">
                      <string>Blocks Overlap Percent:</string>
                     </property>
                     <property name="alignment">
                      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QSpinBox" name="m_blockOverlapPercentSpinBox">
                     <property name="alignment">
                      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                     </property>
                     <property name="maximum">
                      <number>25</number>
                     </property>
                     <property name="value">
                      <number>10</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
               <item row="0" column="1">
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>The default value &quot;0&quot; for &quot;Blocks Overlap Percent&quot; indicates that no merge will be done.</string>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Thread Options</string>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_12">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_11">
             <item row="0" column="0">
              <widget class="QCheckBox" name="m_enableThreadedProcessingcheckBox">
               <property name="toolTip">
                <string>Threaded processing will be performed (best with multi-core or multi-processor systems</string>
               </property>
               <property name="text">
                <string>Enable threaded processing</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <layout class="QGridLayout" name="gridLayout_8">
               <item row="0" column="0">
                <layout class="QGridLayout" name="gridLayout_2">
                 <item row="0" column="0">
                  <widget class="QLabel" name="inputDataSetLabel_5">
                   <property name="minimumSize">
                    <size>
                     <width>170</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Maximum threads number:</string>
                   </property>
                   <property name="alignment">
                    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QLineEdit" name="m_maximumThreadsNumberLineEdit">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="toolTip">
                    <string>The maximum number of concurrent segmenter threads (default:0 - automatically found)</string>
                   </property>
                   <property name="text">
                    <string>0</string>
                   </property>
                   <property name="alignment">
                    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item row="0" column="1">
                <spacer name="horizontalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>The default value &quot;0&quot; indicates that the application will calculate the best value based on your system.</string>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>m_enableThreadedProcessingcheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_maximumBlockSizeLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>172</y>
    </hint>
    <hint type="destinationlabel">
     <x>228</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_enableThreadedProcessingcheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_maximumThreadsNumberLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>68</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
